app/models/container_node.rb

```diff
@@ -75,6 +75,12 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    if MiqServer.active_miq_servers.where(:has_active_cockpit_ws => true)
```

Comment:
The requirement I was given for this was that if there is no server capable of running cockpit-ws we would keep the old behavior. Because users can assign the role to a server without knowing if it has the relevant packages installed I needed to add some extra checks. Hence just checking for the role was not enough. If there is a better way to do this then please let me know.